spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/hazelcast/HazelcastServerConfiguration.java

```diff
@@ -60,10 +63,12 @@ private static HazelcastInstance getHazelcastInstance(Config config) {
 	static class HazelcastServerConfigFileConfiguration {
 
 		@Bean
-		HazelcastInstance hazelcastInstance(HazelcastProperties properties, ResourceLoader resourceLoader)
+		HazelcastInstance hazelcastInstance(HazelcastProperties properties, ResourceLoader resourceLoader,
+											ConfigurationCustomizer customizer)
```

Comment:
Couldn't we use `List<ConfigurationCustomizer> customizers` to avoid the need of `EMPTY_CUSTOMIZER` and at the same time support multiple customizers?

Reply:
This is a good suggestion, thank you. I'm not sure at this point that we need to support multiple beans, but avoiding the need for `EMPTY_CUSTOMIZER` is a good goal. Our style is the achieve that via Spring Framework's `ObjectProvider` and its `getIfAvailable()` method.

Reply:
Good suggestions, thanks to both of you! 

I pushed an update with ObjectProvider and indeed it simplified the code quite a bit. 

I also made the customizer to return `null`. It makes it less flexible, but that's not a concern here. It led to further code simplification and it's also aligned with most of the other customizers I saw in the Spring codebase.